src/components/CheckBoxFilterContainer.ts

```diff
@@ -0,0 +1,187 @@
+interface OfflineConstraint {
+    attribute: string;
+    operator: string;
+    value: string;
+    path?: string;
+}
+
+export interface ContainerState {
+    alertMessage: string;
+    listViewAvailable: boolean;
+    targetListView?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class CheckboxFilterContainer extends Component<ContainerProps, ContainerState> {
+    private dataSourceHelper: DataSourceHelper;
+    private navigationHandler: object;
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { listViewAvailable: false, alertMessage: Utils.validateProps(props) };
+        this.applyFilter = this.applyFilter.bind(this);
+        // Ensures that the listView is connected so the widget doesn't break in mobile due to unpredictable render timing
+        this.navigationHandler = dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView.bind(this));
+    }
+
+    render() {
+        // this.alertMessage = this.validate(this.props, this.state.targetListView) || this.alertMessage || "";
+        return createElement("div",
+            {
+                className: classNames("widget-checkbox-filter", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            this.renderAlert(this.state.alertMessage),
+            this.renderComponent()
+        );
+    }
+
+    componentDidMount() {
+        const filterNode = findDOMNode(this).parentNode as HTMLElement;
+        const targetNode = Utils.findTargetNode(filterNode);
+        DataSourceHelper.hideContent(targetNode);
+    }
+
+    componentDidUpdate(_prevProps: ContainerProps, prevState: ContainerState) {
+        if (this.state.listViewAvailable && !prevState.listViewAvailable) {
+            this.applyFilter(this.props.defaultChecked);
+        }
+    }
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    private renderAlert(message: string) {
+        return createElement(Alert, {
+            className: "widget-checkbox-filter-alert",
+            message
+        });
+    }
+
+    private renderComponent(): ReactElement<CheckboxFilterProps> {
+        if (!this.state.alertMessage) {
+            return createElement(CheckboxFilter, {
+                handleChange: this.applyFilter,
+                isChecked: this.props.defaultChecked
+            });
+        }
+
+        return null;
+    }
+
+    private applyFilter(isChecked: boolean) {
+        const constraint = this.getConstraint(isChecked);
+        if (this.dataSourceHelper) {
+            this.dataSourceHelper.setConstraint(this.props.friendlyId, constraint);
+        }
+    }
+
+    private getConstraint(isChecked: boolean) {
+        const { targetListView } = this.state;
+        if (targetListView && targetListView._datasource) {
+            const attribute = isChecked ? this.props.attribute : this.props.unCheckedAttribute;
```

Comment:
Ternary should be in 3 lines

Reply:
Not necessary these can be inlined. They are short and brief